spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/SourceDataEntriesProcessor.java

```diff
@@ -184,7 +184,7 @@ private static Map<String, Object> extractProperties(String resourceName, String
 			}
 			else {
 				LOG.debug("entry : " + resourceName + " will be treated as a single yml/yaml file");
-				return yamlParserGenerator(environment).andThen(PROPERTIES_TO_MAP).apply(content);
+				return yamlParserGenerator(environment).apply(content);
```

Comment:
`yamlParserGenerator` now returns a `Map<String, Object>` instead of `Properties`, so we can get rid of one step